articles/validating-xml-using-xsd/index.md

```diff
@@ -0,0 +1,262 @@
+
+Syntax:
+```xml
+<xs:attribute name="xxx" type="yyy"/>
+```
+
+##### Restrictions
+When validations are to be done only for one element, it can be done under XSD tag `xs:simpleType`. Here, the validations are more specific to the type for the particular element.
+
+Whereas, when validations are to be done for a set/sequence of elements, it can be specified under `xs:complexType`. Here, XSD validates all the sub-child elements under a parent element, in sequence. More about these restrictions are explained in the below implementation.
+
+#### Validations
+##### XSD Definition
+For any XSD document, the XML declaration statement is important. Along with it, we must also specify namespace details, as shown below:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" />
+```
+
+The first line specifies the version and the encoding type used in the XML document.
+
+The following fragment:
+
+```xml
+xmlns:xs="http://www.w3.org/2001/XMLSchema"
+```
+
+indicates that the elements and data types used in the schema come from the `http://www.w3.org/2001/XMLSchema` namespace. It also specifies that the elements and data types that come from the `http://www.w3.org/2001/XMLSchema` namespace should be prefixed with `xs`.
+
+The following fragment:
+
+```xml
+elementFormDefault="qualified"
+```
+
+indicates that any elements used by the XML instance document which were declared in this schema must be namespace-qualified.
+
+##### Validating the outermost element
+In our XML example, we have an element with `name="root"` as the outermost element holding other child elements. So, in our XSD, we access the tag with `name="root"` and handle the child elements under an `xs:complexType` with `name="rootType"`.
+
+```xml
+<xs:element name="root" type="rootType" />
+```
+
+The above fragment tells us that, whenever a tag with `name="root"` is encountered, the validation type is handled by `name="rootType"`.
+
+Now, we may have any number of `name="event"` tags inside a `name="root"` element, with minimum occurance of `name="event"` tag being `1`. Since, we have atleast 1 `name="event"` element, the definition for `name="event"` tag can be enclosed into a sequence `xs:sequence` as shown below
+
+```xml
+<xs:complexType name="rootType">
+    <xs:sequence minOccurs="0" maxOccurs="unbounded">
+        <xs:element name="event" type="eventType" />
+    </xs:sequence>
+</xs:complexType>
+```
+
+##### Validating child elements
+As we understand, on specifying the `name="event"` element, validating it with a complex type containing a sequence of other sub-child elements under the type called `name="eventType"`. Since, each `name="event"` tag can occur only once, we restrict the limitations of its maximum occurance to `1`.
```

Comment:
> As we understand, on specifying the name="event" element, validating it with a complex type containing a sequence of other sub-child elements under the type called name="eventType".

Breaking this sentence into smaller sentences would help in improved readability.